mta/mta_services.go

```diff
@@ -25,14 +26,105 @@ func createMtaYamlFile(path string) (err error) {
 }
 
 // CreateMta - create MTA project
-func CreateMta(path string, mtaDataJSON string) error {
+func CreateMta(path string, mtaDataJSON string, mkDirs func(string, os.FileMode) error) error {
 	mtaDataYaml, err := yaml.JSONToYAML([]byte(mtaDataJSON))
 	if err != nil {
 		return err
 	}
-	err = createMtaYamlFile(filepath.Join(path))
+	err = createMtaYamlFile(filepath.Join(path), mkDirs)
 	if err != nil {
 		return err
 	}
 	return ioutil.WriteFile(path, mtaDataYaml, 0644)
 }
+
+//AddModule - add new module
+func AddModule(path string, moduleDataJSON string, marshal func(interface{}) ([]byte, error)) error {
+	mtaContent, err := ioutil.ReadFile(filepath.Join(path))
+	if err != nil {
+		return err
+	}
+
+	mtaObj, err := Unmarshal(mtaContent)
+	if err != nil {
+		return err
+	}
+
+	moduleDataYaml, err := yaml.JSONToYAML([]byte(moduleDataJSON))
+	if err != nil {
+		return err
+	}
+
+	module := Module{}
+	err = yaml.Unmarshal(moduleDataYaml, &module)
+	if err != nil {
+		return err
+	}
+
+	mtaObj.Modules = append(mtaObj.Modules, &module)
+
+	mtaBytes, err := marshal(mtaObj)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(path, mtaBytes, 0644)
+}
+
+//AddResource - add new resource
+func AddResource(path string, resourceDataJSON string, marshal func(interface{}) ([]byte, error)) error {
+	mtaContent, err := ioutil.ReadFile(filepath.Join(path))
+	if err != nil {
+		return err
+	}
+
+	mtaObj, err := Unmarshal(mtaContent)
+	if err != nil {
+		return err
+	}
+
+	resourceDataYaml, err := yaml.JSONToYAML([]byte(resourceDataJSON))
+	if err != nil {
+		return err
+	}
+
+	resource := Resource{}
+	err = yaml.Unmarshal(resourceDataYaml, &resource)
+	if err != nil {
+		return err
+	}
+
+	mtaObj.Resources = append(mtaObj.Resources, &resource)
+
+	mtaBytes, err := marshal(mtaObj)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(path, mtaBytes, 0644)
+}
+
+// CopyFile - copy from source path to target path
+func CopyFile(src, dst string, create func(string) (*os.File, error)) (rerr error) {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		rerr = in.Close()
```

Comment:
this way you override errors raised by the method. 
use the following:
e := in.Close()
if rerr == nil {
   rerr = e
}